components/movementsensor/adxl345/adxl345.go

```diff
@@ -232,6 +358,44 @@ func (adxl *adxl345) writeByte(ctx context.Context, register, value byte) error
 	return handle.WriteByteData(ctx, register, value)
 }
 
+func (adxl *adxl345) configureInterruptRegisters(ctx context.Context) error {
+	if adxl.interruptsEnabled == 0 {
+		return nil
+	}
+	adxl.configuredRegisterValues[IntEnableAddr] = adxl.interruptsEnabled
+	adxl.configuredRegisterValues[IntMapAddr] = adxl.interruptsMap
+	for key, value := range defaultRegisterValues {
+		if configuredVal, ok := adxl.configuredRegisterValues[key]; ok {
+			value = configuredVal
+		}
+		err := adxl.writeByte(ctx, key, value)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (adxl *adxl345) readInterrupts(ctx context.Context) {
+	adxl.mu.Lock()
+	defer adxl.mu.Unlock()
+	intSourceRegister, err := adxl.readByte(ctx, IntSourceAddr)
+	if err != nil {
+		adxl.logger.Error(err)
+	}
+
+	for key, value := range interruptBitPosition {
+		if intSourceRegister&value&adxl.interruptsEnabled != 0 {
+			_, ok := adxl.interruptsFound[key]
+			if ok {
+				adxl.interruptsFound[key]++
+			} else {
+				adxl.interruptsFound[key] = 1
```

Comment:
No need for such a complicated way of incrementing the value by 1. Values in a Go map default to 0:
```go
for key, value := range interruptBitPosition {
    if intSourceRegister & value & adxl.interruptsEnabled != 0 {
        adxl.interruptsFound [key]++
    }
}
```